src/rules/terNoMixedSpacesAndTabsRule.ts

```diff
@@ -0,0 +1,137 @@
+import * as ts from 'typescript';
+import * as Lint from 'tslint';
+import { getLineRanges, getTokenAtPosition, isPositionInComment } from 'tsutils';
+
+const RULE_NAME = 'ter-no-mixed-spaces-and-tabs';
+interface ITerNoMixedSpacesAndTabsRuleOptions {
+  readonly tabs?: boolean;
+  readonly smartTabs: boolean;
+}
+
+const OPTION_USE_TABS = 'tabs';
+const OPTION_USE_SPACES = 'spaces';
+
+export class Rule extends Lint.Rules.AbstractRule {
+  public static metadata: Lint.IRuleMetadata = {
+    ruleName: RULE_NAME,
+    description: 'Enforces indentation with unmixed tabs or spaces.',
+    rationale: Lint.Utils.dedent`
+      Using only one of tabs or spaces for indentation leads to more consistent editor behavior,
+      cleaner diffs in version control, and easier programmatic manipulation.`,
+    optionsDescription: Lint.Utils.dedent`
+      This rule takes an object argument with an optional \`type\` property which can be set to:
+
+      * \`${OPTION_USE_SPACES}\` enforces consistent spaces for indentation.
+      * \`${OPTION_USE_TABS}\` enforces consistent tabs for indentation.
+
+      If the above is not provided, the rule will enforce either all tabs or all spaces on each
+      line, although different lines may differ between tabs and spaces.
+
+      Optionally, a \`smartTabs\` boolean property can be specified.  If set to true, smart tabs
+      allow mixing tabs and spaces if tabs are used for indentation and spaces for alignment, eg.
+
+          function main() {
+          // --->const a = 1,
+          // --->......b = 2;
+
+              const a = 1,
+                    b = 2;
+          }
+      `,
+    options: {
+      type: 'array',
+      items: [
+        {
+          type: 'object',
+          properties: {
+            type: {
+              type: 'string',
+              enum: [OPTION_USE_TABS, OPTION_USE_SPACES]
+            },
+            smartTabs: {
+              type: 'boolean'
+            }
+          },
+          additionalProperties: false
+        }
+      ],
+      minLength: 0,
+      maxLength: 1
+    },
+    optionExamples: [
+      Lint.Utils.dedent`
+      "${RULE_NAME}": { type: OPTION_USE_TABS } ]
+      `,
+      Lint.Utils.dedent`
+      "${RULE_NAME}": { type: OPTION_USE_SPACES } ]
```

Comment:
This is what gets placed in the markdown file. Make sure that `type` and its value get double quotes since that is what gets displayed. Also, make sure to use `${}` so that we can see what values `OPTION_USER_SPACES` has.